docs/migration.md

```diff
@@ -0,0 +1,301 @@
+      "element": "array",
+      "content": [
+        {
+          "element": "member",
+          "content": {
+            "key": {
+              "element": "string",
+              "content": "HOST"
+            },
+            "value": {
+              "element": "string",
+              "content": "http://polls.apiblueprint.org/"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Source Map Element
+
+A source map element may contain an optional line and column number to make it
+easier to handle source map information, computing the line and column number
+can often be expensive so it may be provider by a parser. Note however that it
+is optional and it is down the each individual tooling on whether it is
+present, some tools only provide line and column number for source maps
+contained within Annotation Elements.
+
+```json
+{
+  "element": "sourceMap",
+  "content": [
+    {
+      "element": "array",
+      "content": [
+        {
+          "element": "number",
+          "attributes": {
+            "line": {
+              "element": "number",
+              "content": 3
+            },
+            "column": {
+              "element": "number",
+              "content": 2
+            }
+          },
+          "content": 4
+        },
+        {
+          "element": "number",
+          "attributes": {
+            "line": {
+              "element": "number",
+              "content": 3
+            },
+            "column": {
+              "element": "number",
+              "content": 10
+            }
+          },
+          "content": 12
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Data Structure Elements
+
+#### Enumeration Element
+
+The layout of the enum element has been altered. The enumerations have been
+moved to an enumerations attribute of the element and the content represented
+the given value, like in other elements.
+
+Enumerations themselves are an array of the possible enumerations.
+
+Before
+
+```json
+{
+  "element": "enum",
+  "content": [
+    {
+      "element": "string",
+      "content": "north"
+    },
+    {
+      "element": "string",
+      "content": "east"
+    },
+    {
+      "element": "string",
+      "content": "south"
+    },
+    {
+      "element": "string",
+      "content": "west"
+    }
+  ]
+}
+```
+
+After
+
+```json
+{
+  "element": "enum",
+  "attributes": {
+    "enumerations": {
+      "element": "array",
+      "content": [
+        {
+          "element": "string",
+          "content": "north"
+        },
+        {
+          "element": "string",
+          "content": "east"
+        },
+        {
+          "element": "string",
+          "content": "south"
+        },
+        {
+          "element": "string",
+          "content": "west"
+        }
+      ]
+    }
+  }
+}
+```
+
+The intent of the structure was that it represents an enumeration of north,
+east, south and west. As the enumerations do not include a `fixed` type
```

Comment:
Possibly nitpicking, but I thought cardinal points are title cased in English

Reply:
Since I am refering to the JSON values which are lowercase I don't want to mix the spelling. I've wrapped them in code blocks which I think makes that clearer.